applications/DeepSpeed-Chat/training/utils/data/data_utils.py

```diff
@@ -286,6 +287,33 @@ def create_prompt_dataset(local_rank,
             eval_dataset = ConcatDataset(eval_datasets)
             shuffle_idx = get_shuffle_idx(seed, eval_size)
             eval_dataset = Subset(eval_dataset, shuffle_idx.tolist())
+
+        # Append the SFT-only dataset if it exists, and current phase is 1(SFT).
+        if train_phase == 1 and sft_only_data_path:
+            sft_train_datasets = []
+            sft_eval_datasets = []
+            sft_train_size = 0
+            sft_eval_size = 0
+            for sft_path in sft_only_data_path:
+                sft_train_dataset, sft_eval_dataset = create_dataset(
+                    local_rank, sft_path, "10,0,0", output_path, train_phase,
+                    seed, tokenizer, end_of_conversation_token, max_seq_len)
+                sft_train_datasets.append(sft_train_dataset)
+                sft_eval_datasets.append(sft_eval_dataset)
+                sft_train_size += len(sft_train_dataset)
+                sft_eval_size += len(sft_eval_datasets)
+            if sft_train_datasets:  # Check if sft_train_datasets is not empty
+                sft_train_dataset = ConcatDataset(sft_train_datasets)
+                shuffle_idx = get_shuffle_idx(seed, sft_train_size)
```

Comment:
I recommend to first do train_dataset = ConcatDataset([train_dataset, sft_train_dataset]) and then shuffle the whole train_dataset, otherwise your train data will be not blended well enough.